Algorithms-Big-O-Notation.md

```diff
@@ -0,0 +1,54 @@
+# Big-O Notation
+
+In mathematics, Big-O notation is a symbolism used to describe and compare the limiting behavior of a function.
+In short Big-O notation is used to describe the growth or decline of a function, usually with respect to another function.
+For example we say that x = O(x^2) for all x > 1 or in other words, x^2 is an upper bound on x and therefore it grows faster.
```

Comment:
You might want to toss in a line describing what you mean by the `^` symbol. In this case, it is exponentiation, but in  few common programming languages, its XOR.